---
title: "Figure 4. ATAC-seq analysis of C9ORF50 knockout effect."
author: "Chuanyang Liu, Lvyun Zhu"
date: "2023/10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# Clean workspace
rm(list=ls())
# Set output directory
output_dir="./"
```

## ATAC-seq Peak Annotation Analysis

This section performs peak annotation analysis using ChIPseeker package to understand the genomic distribution of ATAC-seq peaks in C9ORF50 knockout and control samples.

```{r peak-annotation, echo=TRUE}
setwd("B:/Kinda_Help/Lvyun_Zhu-C9orf50/ATAC-seq/")

library(ChIPseeker)
library(ggplot2)
library(GenomicFeatures)
library(txdbmaker)

# Create TxDb objects from GFF files
mm <- makeTxDbFromGFF("gene.gff3")
mm2 <- makeTxDbFromGFF("genecode_self_v43.gff3")

# Read peak files
C9KO1 <- readPeakFile("Data/C9orf50KO1_summits.bed")
C9KO2 <- readPeakFile("Data/C9orf50KO2_summits.bed")
C9KO3 <- readPeakFile("Data/C9orf50KO3_summits.bed")
C1 <- readPeakFile("Data/Control1_summits.bed")
C2 <- readPeakFile("Data/Control2_summits.bed")
C3 <- readPeakFile("Data/Control3_summits.bed")

# Annotate peaks
C9KO1_peakAnno <- annotatePeak(C9KO1, tssRegion = c(-3000, 3000), TxDb=mm2)
C9KO2_peakAnno <- annotatePeak(C9KO2, tssRegion = c(-3000, 3000), TxDb=mm2)
C9KO3_peakAnno <- annotatePeak(C9KO3, tssRegion = c(-3000, 3000), TxDb=mm2)
C1_peakAnno <- annotatePeak(C1, tssRegion = c(-3000, 3000), TxDb=mm2)
C2_peakAnno <- annotatePeak(C2, tssRegion = c(-3000, 3000), TxDb=mm2)
C3_peakAnno <- annotatePeak(C3, tssRegion = c(-3000, 3000), TxDb=mm2)

# Plot annotation distribution
plotAnnoPie(C3_peakAnno)

# Create list of peaks for batch processing
peaks <- list(C9orf50KO1_peakAnno=C9KO1, 
              C9orf50KO2_peakAnno=C9KO2,
              C9orf50KO3_peakAnno=C9KO3,
              Ctl1_peakAnno=C1,
              Ctl2_peakAnno=C2,
              Ctl3_peakAnno=C3)

# Annotate all peaks
peaksAnno <- lapply(peaks, annotatePeak, tssRegion = c(-3000, 3000), TxDb=mm2)

# Plot bar chart of annotation
plotAnnoBar(c(C9KO1_peakAnno, C9KO2_peakAnno, C9KO3_peakAnno,
              C1_peakAnno, C2_peakAnno, C3_peakAnno))

# Save the plot
ggsave("peakAnno.svg", height = 3, width = 6, dpi = 600)

# Save workspace
save(list = ls(), file = "ChIPseeker.rdata")
```

## Peak Visualization

This section visualizes peak distribution using various plots to understand the relationship between peaks in different samples.

```{r peak-visualization, echo=TRUE}
# Load saved data if needed
# load("ChIPseeker.rdata")

# Venn pie chart for control sample
vennpie(C1_peakAnno)

# Upset plot for knockout sample
upsetplot(C9KO1_peakAnno, vennpie=TRUE)

# Get promoter regions
promoter <- getPromoters(TxDb=mm, upstream=3000, downstream=3000)

# Plot average profile
plotAvgProf(peaksAnno, xlim=c(-3000, 3000),
            xlab="Genomic Region (5'->3')", ylab = "Read Count Frequency")
```

## Differential Binding Analysis

This section performs differential binding analysis using DiffBind package to identify significant differences in chromatin accessibility between C9ORF50 knockout and control samples.

```{r diffbind-analysis, echo=TRUE}
rm(list=ls())
setwd("B:/Kinda_Help/Lvyun_Zhu-C9orf50/ATAC-seq/")

library(ChIPseeker)
library(ggplot2)
library(DiffBind)

## Step 1: Read files
# Read sample sheet
datinfo <- read.csv("SampleSheetForDiffBind2.csv", stringsAsFactors = F)
str(datinfo)

# Create DiffBind object
dbObj <- dba(sampleSheet = datinfo)
plot(dbObj)
save(dbObj, file = "DiffBindObject.rdata")

# Count reads
dbObj <- dba.count(dbObj, bUseSummarizeOverlaps = T)
save(dbObj, file = "DiffBindObject2.rdata")
plot(dbObj)

# PCA plot
dba.plotPCA(dbObj)

## Step 2: Differential analysis
# Create contrast
dbObj_contrast <- dba.contrast(dbObj)
save(dbObj, dbObj_contrast, file = "Step2.DiffBindObject.rdata")

# Analyze using all methods
dbObj_contrast <- dba.analyze(dbObj_contrast, method = DBA_ALL_METHODS)
plot(dbObj_contrast)

# MA plot
dba.plotMA(dbObj_contrast)

# Get DESeq2 results
dbObj_DiffResults <- dba.report(dbObj_contrast, method=DBA_DESEQ2, bUsePval = T, th = 0.05)

# Convert to data frame and save
out <- as.data.frame(dbObj_DiffResults)
write.table(out, file="DiffBind_deseq2.txt", sep="\t", quote=F, col.names = NA)

# Volcano plot
dba.plotVolcano(dbObj_DiffResults)
```

## Profile Analysis and Visualization

This section generates profile plots and Venn diagrams to compare chromatin accessibility patterns between samples.

```{r profile-analysis, echo=TRUE}
library(profileplyr)

# Generate profiles
profiles <- dba.plotProfile(dbObj_contrast)
dba.plotProfile(profiles, scores = NULL)
ggsave("Compare.svg", height = 5, width = 3.4, dpi = 600)

# Merge by tissue
profile2 <- dba.plotProfile(dbObj_contrast, merge = DBA_TISSUE)
dba.plotProfile(profile2, scores = NULL)
ggsave("Compare2.svg", height = 5, width = 3.4, dpi = 600)

# Merge by factor
profile3 <- dba.plotProfile(dbObj_contrast, merge = DBA_FACTOR)
dba.plotProfile(profile3, scores = NULL, merge = DBA_CONDITION)
ggsave("Compare3.svg", height = 5, width = 3.4, dpi = 600)

# Venn diagrams
library(VennDiagram)
dba.plotVenn(dbObj_contrast, contrast = 1, method=DBA_ALL_METHODS, bDB = T)
dba.plotVenn(dbObj)

# Save significant peaks in BED format
out <- as.data.frame(dbObj_DiffResults)
deseq.bed <- out[which(out$p.value < 0.05), 
                 c("seqnames", "start", "end", "strand", "Fold")]
write.table(deseq.bed, file="C9KO_vs_CTL_deseq2_sig.bed", sep="\t", 
            quote=F, row.names=F, col.names=F)

# Save workspace
save(profile2, profile3, profiles, deseq.bed, dbObj_DiffResults, 
     dbObj, dbObj_contrast, file = "Step2.DiffBind.rdata")
```

## Peak Annotation of Differential Regions

This section annotates the differentially accessible regions to understand their genomic context.

```{r diff-peak-annotation, echo=TRUE}
# Read significant differential peaks
peakself <- readPeakFile("C9KO_vs_CTL_deseq2_sig.bed") 

# Annotate differential peaks
peakAnno <- annotatePeak(peakself, tssRegion = c(-3000, 3000), TxDb = mm)

# Output results
write.table(peakAnno, file = 'peakDiff.txt', sep = '\t', quote = FALSE, row.names = FALSE)
```

---

*This analysis is part of the paper: "Genome-wide CRISPR screen reveals an uncharacterized spliceosome regulator as a potential immunotherapeutic target for cancer therapy"*